File: tests/fixtures/python/derived.py
"""Derived class inheriting from BaseClass."""

from base import BaseClass


class DerivedClass(BaseClass):
    """Derived class that extends BaseClass."""

    def __init__(self, name: str, id: int):
        super().__init__(name)
        self.id = id

    def get_name(self) -> str:
        """Override get_name to include ID."""
        return f"{self.name} (ID: {self.id})"

    def get_id(self) -> int:
        """Get the ID of the object."""
        return self.id

    def process_data(self, a: int, b: int) -> int:
        """Process data using parent's calculate method."""
        result = self.calculate(a, b)
        return result * self.id

File: tests/fixtures/typescript/derived.ts
/**
 * Derived class inheriting from BaseClass.
 */

import { BaseClass } from './base';

export class DerivedClass extends BaseClass {
    private id: number;

    constructor(name: string, id: number) {
        super(name);
        this.id = id;
    }

    /**
     * Override getName to include ID.
     */
    getName(): string {
        return `${this.name} (ID: ${this.id})`;
    }

    /**
     * Get the ID of the object.
     */
    getId(): number {
        return this.id;
    }

    /**
     * Process data using parent's calculate method.
     */
    processData(a: number, b: number): number {
        const result = this.calculate(a, b);
        return result * this.id;
    }
}
